saleor/graphql/product/bulk_mutations/product_bulk_create.py

```diff
@@ -0,0 +1,832 @@
+    category = graphene.ID(description="ID of the product's category.", name="category")
+    collections = NonNullList(
+        graphene.ID,
+        description="List of IDs of collections that the product belongs to.",
+        name="collections",
+    )
+    description = JSONString(description="Product description." + RICH_CONTENT)
+    name = graphene.String(description="Product name.")
+    slug = graphene.String(description="Product slug.")
+    tax_class = graphene.ID(
+        description=(
+            "ID of a tax class to assign to this product. If not provided, product "
+            "will use the tax class which is assigned to the product type."
+        ),
+        required=False,
+    )
+    seo = SeoInput(description="Search engine optimization fields.")
+    weight = WeightScalar(description="Weight of the Product.", required=False)
+    rating = graphene.Float(description="Defines the product rating value.")
+    metadata = NonNullList(
+        MetadataInput,
+        description="Fields required to update the product metadata.",
+        required=False,
+    )
+    private_metadata = NonNullList(
+        MetadataInput,
+        description=("Fields required to update the product private metadata."),
+        required=False,
+    )
+    external_reference = graphene.String(
+        description="External ID of this product.", required=False
+    )
+    product_type = graphene.ID(
+        description="ID of the type that product belongs to.",
+        name="productType",
+        required=True,
+    )
+    media = NonNullList(MediaInput, required=False)
+    channel_listings = NonNullList(ProductChannelListingCreateInput, required=False)
+    variants = NonNullList(
+        ProductVariantBulkCreateInput,
+        required=False,
+        description="Input list of product variants to create.",
+    )
+
+
+class ProductBulkCreate(BaseMutation):
+    count = graphene.Int(
+        required=True,
+        description="Returns how many objects were created.",
+    )
+    results = NonNullList(
+        ProductBulkResult,
+        required=True,
+        default_value=[],
+        description="List of the created products.",
+    )
+
+    class Arguments:
+        products = NonNullList(
+            ProductBulkCreateInput,
+            required=True,
+            description="Input list of products to create.",
+        )
+        error_policy = ErrorPolicyEnum(
+            required=False,
+            default_value=ErrorPolicyEnum.REJECT_EVERYTHING.value,
+            description="Policies of error handling.",
+        )
+
+    class Meta:
+        description = "Creates products." + ADDED_IN_313 + PREVIEW_FEATURE
+        permissions = (ProductPermissions.MANAGE_PRODUCTS,)
+        error_type_class = ProductBulkCreateError
+        support_meta_field = True
+        support_private_meta_field = True
+
+    @classmethod
+    def generate_unique_slug(cls, slugable_value):
```

Comment:
After your comment, I had to add extra logic to this function, we need to build a list of slugs for new products because we can have in one bulk mutation 10 products with the same name and `saleor.core.utils.generate_unique_slug` will create same slug for all of them